lib/travis_dedup.rb

```diff
@@ -98,17 +100,36 @@ def host
     end
 
     def request(method, path, params, headers={})
+      attempts = 0
+      begin
+        faraday_send(method, path, params, headers)
+      rescue TravisError500 => error
+        raise error if (attempts += 1) > max_attempts
+        retry
+      end
+    end
+
+    def max_attempts
+      DEFAULT_MAX_ATTEMPTS_WHEN_TRAVIS_DOES_NOT_RESPOND
```

Comment:
next commit was going to introduce a `--retry` option and change this method. Will amend it.